addons/mail/static/src/views/fields/badge_selection_icons/badge_selection_icons_field.js

```diff
@@ -0,0 +1,57 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { getFieldDomain } from "@web/model/relational_model/utils";
+import { useSpecialData } from "@web/views/fields/relational_utils";
+import { badgeSelectionField, BadgeSelectionField } from "@web/views/fields/badge_selection/badge_selection_field";
+
+
+export class BadgeSelectionWithIconsField extends BadgeSelectionField {
+    /**
+     * @typedef BadgeSelectionIconsField
+     * Overrides the standard BadgeSelectionField and inserts FontAwesome icons before each option's title.
+     * Only compatible with Many2one selectors. Related options should have an "icon" field, the name
+     * of which should be specified through the iconField prop.
+     * 
+     * Special props:
+     * @param {String} iconField The name of the field on which the icon is stored on the many2one option
+     * @param {String} defaultIcon If the field pointed through iconField is empty on the related record, a default fa icon can be specified.
+     */
+    static props = {
+        ...BadgeSelectionField.props,
+        iconField: { type: String, },
+        defaultIcon: { type: String, optional: true, default: "fa-check"}
+    };
+    static template="mail.BadgeSelectionIconsField";
+
+    async setup() {
+        this.type = this.props.record.fields[this.props.name].type;
```

Comment:
Note that not calling super is really not great.

Ideally the base component should be modified, probably to make "specialData" an overridable getter.
But we don't have much time so leave it like this.

I would however explain in the docstring that "specialData" in the parent component expects the result of a "name_search" so you have to manually adapt the result of "search_read" to make it fit into a list of list instead of a list of objects (if I understood properly).